contracts/token/ERC20/EncryptedERC20.sol

```diff
@@ -1,201 +1,182 @@
 // SPDX-License-Identifier: BSD-3-Clause-Clear
-
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.24;
 
 import "fhevm/lib/TFHE.sol";
-import "fhevm/abstracts/Reencrypt.sol";
-import "../../utils/EncryptedErrors.sol";
+import { IEncryptedERC20 } from "./IEncryptedERC20.sol";
 
-abstract contract EncryptedERC20 is Reencrypt, EncryptedErrors {
-    enum ErrorCodes {
-        NO_ERROR,
-        UNSUFFICIENT_BALANCE,
-        UNSUFFICIENT_APPROVAL
-    }
+/**
```

Comment:
Please add more detailed explanations in the natspecs, you can see the work that @poppyseedDev already did on fhevm example: https://github.com/zama-ai/fhevm/blob/f821ca70e3cd6880d4d4ec55fc2196d8b6ccf132/examples/EncryptedERC20.sol#L8-L9